fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/recordrepair/RecordValidateOnlyTest.java

```diff
@@ -162,73 +117,52 @@ void testValidateMissingSplit(int splitNumber, FormatVersion formatVersion, bool
             final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
             storeBuilder = store.asBuilder();
         }
-        RecordRepairRunner runner = RecordRepairRunner.builder(fdb).build();
-        RecordValidationStatsResult repairStats = runner.runValidationStats(storeBuilder, validationKind);
-        List<RecordValidationResult> repairResults = runner.runValidationAndRepair(storeBuilder, validationKind, false);
-
-        if (splitNumber == 0) {
-            if (storeVersions) {
-                if (ValidationTestUtils.versionStoredWithRecord(formatVersion)) {
-                    // record split gone but version remains
-                    Assertions.assertThat(repairStats.getStats()).hasSize(1);
-                    Assertions.assertThat(repairStats.getStats()).containsEntry(RecordValueValidator.CODE_SPLIT_ERROR, 1);
-                    Assertions.assertThat(repairResults).hasSize(1);
-                    Assertions.assertThat(repairResults.get(0)).isEqualTo(RecordValidationResult.invalid(primaryKey, RecordValueValidator.CODE_SPLIT_ERROR, "any"));
+        RecordRepair.Builder builder = RecordRepair.builder(fdb, storeBuilder).withValidationKind(validationKind);
+        try (RecordRepairStatsRunner statsRunner = builder.buildStatsRunner();
+                RecordRepairValidateRunner repairRunner = builder.buildRepairRunner(false)) {
+            RepairStatsResults repairStats = statsRunner.run().join();
+            RepairValidationResults repairResults = repairRunner.run().join();
+            List<RecordRepairResult> invalidResults = repairResults.getInvalidResults();
+
+            if (splitNumber == 0) {
+                if (storeVersions) {
+                    if (ValidationTestUtils.versionStoredWithRecord(formatVersion)) {
+                        // record split gone but version remains
+                        ValidationTestUtils.assertRepairStats(repairStats, NUM_RECORDS - 1, 1, RecordRepairResult.CODE_SPLIT_ERROR);
+                        ValidationTestUtils.assertCompleteResults(repairResults, NUM_RECORDS);
+                        ValidationTestUtils.assertInvalidResults(
+                                invalidResults,
+                                1,
+                                result -> result.equals(RecordRepairResult.invalid(primaryKey, RecordRepairResult.CODE_SPLIT_ERROR, "any")));
```

Comment:
You could have
```
ValidationTestUtils.assertResults(repairStats, repairResults, numRecords, 1, RecordRepairResult.CODE_SPLIT_ERROR, result -. result.getPrimaryKey().equals(primaryKey));
```

I think. Although, I'm not sure about the primary key.
A different one for `ValidationTestUtils.assertAllValid(repairStats, repairResults, NUM_RECORDS -1);` might be simpler, and useful too.

Reply:
Note: for this test specifically, you could also reduce a lot of the repetition by having:
```
if (splitNumber != 0 || (storeVersions && !ValidationTestUtils.versionStoredWithRecord(formatVersion))) {
    expectedErrorCode = ....
    ...
} else {
    // assert all valid `NUM_RECORDS - 1`
}
```

But that complex if block is maybe less necessary if the bodies are simplified.

Reply:
I guess for the primaryKey aspect, you could actually follow a pattern more similar to the end of `testValidateRecordsMissingVersion`, where you assert about primary keys completely independently.